### PR TITLE
View job roles UI test

### PR DESCRIPTION
**Pull request by henrykainos**

### CODE REVIEW

File: src/test/ui/ChromeDriver.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-import { Builder, By, WebDriver } from 'selenium-webdriver';
+import { Builder, By, WebDriver, WebElement } from 'selenium-webdriver';
 
 export class ChromeDriver {
     driver: WebDriver;
 
     constructor(driver?: WebDriver) {
         const options = new chrome.Options();
-            options.addArguments('headless'); // Ensure Chrome is running in headless mode
+            //options.addArguments('headless'); // Ensure Chrome is running in headless mode
```
Comment (shaunganley) [inline src/test/ui/ChromeDriver.ts]: should this be uncommented?

File: src/test/ui/ChromeDriver.ts [hunk 2]
```diff
@@ -42,6 +42,53 @@ export class ChromeDriver {
         return await this.driver.findElement(By.linkText(linkText)).getText();
     }
 
+    async getElementById(id: string) {
+        console.log(`Getting element with ID: ${id}`);
+        return await this.driver.findElement(By.id(id))
+    }
+
+    async getElementByXpath(path: string) {
```
Comment (shaunganley) [inline src/test/ui/ChromeDriver.ts:53]: don't get anything by xpath, delete

File: src/test/ui/ChromeDriver.ts [hunk 3]
```diff
@@ -42,6 +42,53 @@ export class ChromeDriver {
...
+        console.log(`Getting element with XPath: ${path}`);
+        return await this.driver.findElement(By.xpath(path));
+    }
+
+    async getElementsByCSS(CSS: string) {
+        console.log(`Getting element with CSS: ${CSS}`);
+        return await this.driver.findElements(By.css(CSS));
+    }
+
+    async getTableById(id: string) {
```
Comment (shaunganley) [inline src/test/ui/ChromeDriver.ts]: this is a duplicate of getElementById method

File: src/test/ui/ChromeDriver.ts [hunk 4]
```diff
@@ -42,6 +42,53 @@ export class ChromeDriver {
...
+    async getElementsByCSS(CSS: string) {
+        console.log(`Getting element with CSS: ${CSS}`);
+        return await this.driver.findElements(By.css(CSS));
+    }
+
+    async getTableById(id: string) { 
+        return await this.driver.findElement(By.id(id));
+    }
+
+    async getTableWithXpath(xpath: string): Promise<WebElement> {
```
Comment (shaunganley) [inline src/test/ui/ChromeDriver.ts]: no xpath

File: src/test/ui/JobRolesTestPage.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+
+export class JobRolesTestPage extends ChromeDriver {
+
+    constructor(driver: WebDriver) {
+        super(driver);
+    }
+
+    async open(): Promise<void> {
+        const WEBSITE_URL = process.env.WEBSITE_URL;
```
Comment (shaunganley) [inline src/test/ui/JobRolesTestPage.ts]: have a default for local host if the url isn't set

**Review by shaunganley — CHANGES_REQUESTED** [review]

File: src/test/ui/ChromeDriver.ts (hunk 2 above)
Comment (henrykainos) [reply src/test/ui/ChromeDriver.ts:53]: I use it as placeholder until ID's get implemented

**Review by shaunganley — APPROVED** [review]

**Review by kajatraczykowska — APPROVED** [review]

**Pull request merged**